src/main/java/org/isf/stat/manager/JasperReportsManager.java

```diff
@@ -773,6 +854,91 @@ public void getGenericReportMYExcel(Integer month, Integer year, String jasperFi
+		 * encoding. For this reason we pass also the resource for the specific
+		 * report read with UTF8Control()
+		 */
+		parameters.put("REPORT_RESOURCE_BUNDLE", getReportResourceBundle(jasperFileName, GeneralData.LANGUAGE));
+
+		/*
+		 * Jasper Reports may contain subreports and we should pass also those.
+		 * The parent report must contain parameters like:
+		 * 
+		 * SUBREPORT_RESOURCE_BUNDLE_1 
+		 * SUBREPORT_RESOURCE_BUNDLE_2
+		 * SUBREPORT_RESOURCE_BUNDLE_...
+		 * 
+		 * and pass them as REPORT_RESOURCE_BUNDLE to each related subreport.
+		 * 
+		 * If nothing is passed, subreports still work, but REPORT_LOCALE will be used 
+		 * (if passed to the subreport) and corresponding bundle (UTF-8 decoding not available) 
+		 */
+		try {
+			addSubReportsBundleParameters(jasperFileName, parameters);
+		} catch (JRException e) {
+			logger.error(">> error loading subreport bundle, default will be used");
+			logger.error(e.getMessage());
+		}
+	}
+	
+	private void addSubReportsBundleParameters(String jasperFileName, HashMap<String, Object> parameters) throws JRException {
+		logger.debug("Search subreports for {}...", jasperFileName);
+		File jasperFile = new File(compileJasperFilename(jasperFileName));
+		final JasperReport jasperReport = (JasperReport) JRLoader.loadObject(jasperFile);
+		JRBand[] bands = jasperReport.getAllBands(); // Get all bands
+		for (JRBand band : bands) {
+			List<JRChild> elements = band.getChildren(); // Get all children
+			for (JRChild child : elements) {
+				int index = 1;
+				if (child instanceof JRBaseSubreport) { // This is a subreport
+					JRBaseSubreport subreport = (JRBaseSubreport) child;
+					String expression = ""; // Lets find out the expression used
+					JRExpressionChunk[] chunks = subreport.getExpression().getChunks();
+					for (JRExpressionChunk c : chunks) {
+						expression += c.getText();
+					}
+					logger.debug("...found a subreport: {}", expression);
+					addSubreportParameter(index, expression, parameters);
+				}
+			}
+		}
+	}
+
+	private void addSubreportParameter(int index, String expression, HashMap<String, Object> parameters) {
+		Pattern pattern = Pattern.compile("\"(.*)\"");
+		Matcher matcher = pattern.matcher(expression);
+		if (matcher.find()) {
+			String subreportName = matcher.group(1).split("\\.")[0];
+			parameters.put("SUBREPORT_RESOURCE_BUNDLE_" + index, getReportResourceBundle(subreportName, GeneralData.LANGUAGE));
+		} else {
+			logger.debug(">> unexpected subreport expression {}", expression);
+		}
+	}
+
+	private ResourceBundle getReportResourceBundle(String jasperFileName, String language) {
+		ResourceBundle resourceBundle;
+		try {
+			resourceBundle = ResourceBundle.getBundle(
+					jasperFileName, 
+					new Locale(language), 
+					new UTF8Control());
+		} catch (MissingResourceException e) {
+			logger.error(">> no resource bundle for language = {} found for this report.", language);
```

Comment:
would it make sense to add the jasperFileName to the logging message?

Reply:
From my perspective yes, it could be useful, @mwithi what do you thing ?

Reply:
yep!